pkg/noderesourcetopology/cache/overallocating.go

```diff
@@ -0,0 +1,238 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	k8scache "k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+
+	topologyv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+	listerv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/listers/topology/v1alpha1"
+
+	"sigs.k8s.io/scheduler-plugins/pkg/noderesourcetopology/stringify"
+
+	"github.com/k8stopologyawareschedwg/podfingerprint"
+)
+
+type NodeIndexer interface {
+	PodNamespacedNamesByNode(logKey, nodeName string) ([]types.NamespacedName, error)
+}
+
+type Overallocating struct {
+	lock             sync.Mutex
+	nrts             nrtStore
+	assumedResources map[string]resourceStore // nodeName -> resourceStore
+	nodeDiscarded    counter
+	lister           listerv1alpha1.NodeResourceTopologyLister
+	indexer          NodeIndexer
+}
+
+func NewOverallocating(lister listerv1alpha1.NodeResourceTopologyLister, indexer NodeIndexer) (*Overallocating, error) {
+	if lister == nil || indexer == nil {
+		return nil, fmt.Errorf("nrtcache: received nil references")
+	}
+
+	nrtObjs, err := lister.List(labels.NewSelector())
+	if err != nil {
+		return nil, err
+	}
+
+	klog.V(3).InfoS("nrtcache: initializing", "objects", len(nrtObjs))
+	return &Overallocating{
+		nrts:             newNrtStore(nrtObjs),
```

Comment:
This is a interesting take. This design stem by the fact we started with the less intrusive change form the internal API perspective and we kept the NRT data as focal point. But it can totally work the way you are describing, keeping track of pods and invalidating them when the latest NRT matches.

The problem here is to compute the pod fingerprinting, which has to be done at each get attempt. This is potentially costly, 
and this is why we added the resync loop - to throttle and control how many pod fingerprinting we need to do.

I guess we can start with the simplest design and optimize later, then